### PR TITLE
Keep link reference definitions verbatim during wrap (#292)

### LINKED ISSUE #292: Reflow mangles Markdown link reference definitions
## Description

`mdtablefix --wrap` incorrectly treats Markdown link reference definitions as prose paragraphs, collapsing and re-wrapping them. Each link reference definition occupies exactly one logical line and must not be reflowed at all.

## Example

**Original:**

```
[ansible]: <https://docs.ansible.com/projects/ansible/latest/playbook_guide/playbooks_conditionals.html>
[bazel]: <https://bazel.build/docs/configurable-attributes>
[cargo-make]: <https://sagiegurari.github.io/cargo-make/>
[github-actions]: <https://docs.github.com/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow>
[gnu-make]: <https://web.mit.edu/gnu/doc/html/make_7.html>
[just]: <https://just.systems/man/en/conditional-expressions.html>
[taskfile]: <https://taskfile.dev/docs/guide>
```

**Reflowed:**

```
[ansible]:
<https://docs.ansible.com/projects/ansible/latest/playbook_guide/playbooks_conditionals.html>
 [bazel]: <https://bazel.build/docs/configurable-attributes> [cargo-make]:
<https://sagiegurari.github.io/cargo-make/> [github-actions]:
<https://docs.github.com/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow>
 [gnu-make]: <https://web.mit.edu/gnu/doc/html/make_7.html> [just]:
<https://just.systems/man/en/conditional-expressions.html> [taskfile]:
<https://taskfile.dev/docs/guide>
```

The reflow:
1. Collapses all seven single-line definitions into one paragraph.
2. Splits the `[ansible]:` label from its URL across two lines.
3. Introduces spurious leading spaces at the start of certain output lines (e.g., before `[bazel]:` and `[gnu-make]:`).

## Expected behaviour

Link reference definitions (lines matching `[label]: <url>` or `[label]: url`) must be recognised as a distinct block type and left entirely untouched by the reflow pass.

## Notes

- Related to #291, which also documents the spurious leading-space symptom and the incorrect collapsing of this same block type.
- Reported in PR #280 by @leynos.

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch keeps CommonMark link reference definitions verbatim during the
`--wrap` reflow pass. The wrap pipeline now recognises `[label]: url` lines as
a distinct block type and routes them through passthrough handling so they are
not collapsed into prose paragraphs or split across lines.

Closes #292.

## Review walkthrough

- Start with [src/wrap/block.rs](https://github.com/leynos/mdtablefix/blob/b6d42debca33da2ff20c71abbd44923488e9c9d6/src/wrap/block.rs#L42-L115) for the `LINK_REF_RE` regex, `BlockKind::LinkReferenceDefinition`, and classification order after footnote definitions.
- Then review [src/wrap.rs](https://github.com/leynos/mdtablefix/blob/b6d42debca33da2ff20c71abbd44923488e9c9d6/src/wrap.rs#L60-L68) to confirm link reference definitions are passthrough blocks.
- Finish with [tests/wrap/link_reference_definitions.rs](https://github.com/leynos/mdtablefix/blob/b6d42debca33da2ff20c71abbd44923488e9c9d6/tests/wrap/link_reference_definitions.rs) for regression coverage of single definitions, consecutive collections, mixed paragraphs, and optional titles.

## Validation

- `make check-fmt`: passed
- `make lint`: passed
- `make test`: passed
- `coderabbit review --agent`: 0 findings

## Summary by Sourcery

Preserve CommonMark link reference definitions verbatim during the wrapping pass so they are not reflowed as prose.

New Features:
- Detect link reference definition lines as a distinct block kind during wrapping and treat them as passthrough blocks.

Tests:
- Add regression tests ensuring single, multiple, and mixed link reference definitions remain unchanged by wrapping, including cases with bare URLs, titles, and document boundaries.

### CODE REVIEW

File: src/wrap.rs [hunk 1]
```diff
@@ -61,7 +61,11 @@ fn is_passthrough_block(line: &str) -> bool {
     is_table_or_separator(line)
         || matches!(
             classify_block(line),
-            Some(BlockKind::Heading | BlockKind::MarkdownlintDirective)
+            Some(
+                BlockKind::Heading
+                    | BlockKind::MarkdownlintDirective
+                    | BlockKind::LinkReferenceDefinition,
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/wrap.rs:69]: @coderabbitai Please suggest a fix for the this concern and provide an AI coding agent prompt for the fix

[the remark this answers is not in the document]
Comment (leynos) [reply src/wrap.rs:69]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (3 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| User-Facing Documentation | ⚠️ Warning | PR adds user-facing `--wrap` behaviour change (preserving link reference definitions) but does not update docs/users-guide.md or CHANGELOG.md to document the fix. | Update docs/users-guide.md to document that link reference definitions are preserved verbatim during wrapping; add entry to CHANGELOG.md under Fixed section referencing issue `#292`. |
| Developer Documentation | ⚠️ Warning | Link reference definition support was added to wrap module without updating developers-guide.md. New block kind, regexes, and helper functions lack documentation. | Update docs/developers-guide.md to document LinkReferenceDefinition block kind, link\_ref\_needs\_title state tracking, and regex limitations. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces invariants over regex-matchable inputs (various URL formats, title styles, whitespace) but lacks property-based testing for regex correctness and edge cases. | Add proptest tests validating: (1) all valid CommonMark link reference definitions match LINK\_REF\_RE; (2) invalid forms rejected; (3) title capture groups correctly identify optional titles. |

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (6 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| User-Facing Documentation | ⚠️ Warning | PR fixes user-visible wrap behaviour for link reference definitions but does not update docs/users-guide.md or README.md to document this change. | Add documentation to docs/users-guide.md explaining that link reference definitions are preserved verbatim during wrapping, alongside the existing list of other preserved block types. |
| Developer Documentation | ⚠️ Warning | New internal APIs added to src/wrap/block.rs are not documented in docs/developers-guide.md developers guide. | Update docs/developers-guide.md to document the new LinkReferenceDefinition block type and helper functions link\_ref\_needs\_title and is\_link\_title\_line in the Wrap module architecture section. |
| Testing (Unit And Behavioural) | ⚠️ Warning | State machine edge cases missing: no explicit tests for awaiting\_link\_title=true when fence line, non-title paragraph, or in-fence context encountered. | Add unit tests for awaiting\_link\_title state transitions: fence line encounter, non-title paragraph line, and in-fence context paths to verify state clearing. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces invariants (regex patterns, state transitions, preservation) over input ranges that lack property-based testing with proptest to verify correctness universally. | Add proptest tests verifying regex patterns handle arbitrary inputs, state machine transitions complete, and preservation holds for generated link definitions. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | The PR modifies text-based Markdown output; snapshot tests should be recommended for verifiable wrapping behaviour, but only direct assertions are provided. | Add snapshot tests with insta covering representative link reference definition preservation cases to encode meaningful, focused behavioural expectations. |
| Performance And Resource Use | ⚠️ Warning | classify\_block called twice per LinkReferenceDefinition line: once in is\_passthrough\_block and again in the matches! block, redundantly invoking LINK\_REF\_RE.is\_match. | Cache classify\_block result before is\_passthrough\_block check and reuse in the LinkReferenceDefinition branch to eliminate duplicate regex evaluation. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (6 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| User-Facing Documentation | ⚠️ Warning | PR fixes user-visible wrap behaviour for link reference definitions but does not update docs/users-guide.md or README.md to document this change. | Add documentation to docs/users-guide.md explaining that link reference definitions are preserved verbatim during wrapping, alongside the existing list of other preserved block types. |
| Developer Documentation | ⚠️ Warning | New internal APIs added to src/wrap/block.rs are not documented in docs/developers-guide.md developers guide. | Update docs/developers-guide.md to document the new LinkReferenceDefinition block type and helper functions link\_ref\_needs\_title and is\_link\_title\_line in the Wrap module architecture section. |
| Testing (Unit And Behavioural) | ⚠️ Warning | State machine edge cases missing: no explicit tests for awaiting\_link\_title=true when fence line, non-title paragraph, or in-fence context encountered. | Add unit tests for awaiting\_link\_title state transitions: fence line encounter, non-title paragraph line, and in-fence context paths to verify state clearing. |
| Testing (Property / Proof) | ⚠️ Warning | PR introduces invariants (regex patterns, state transitions, preservation) over input ranges that lack property-based testing with proptest to verify correctness universally. | Add proptest tests verifying regex patterns handle arbitrary inputs, state machine transitions complete, and preservation holds for generated link definitions. |
| Testing (Compile-Time / Ui) | ⚠️ Warning | The PR modifies text-based Markdown output; snapshot tests should be recommended for verifiable wrapping behaviour, but only direct assertions are provided. | Add snapshot tests with insta covering representative link reference definition preservation cases to encode meaningful, focused behavioural expectations. |
| Performance And Resource Use | ⚠️ Warning | classify\_block called twice per LinkReferenceDefinition line: once in is\_passthrough\_block and again in the matches! block, redundantly invoking LINK\_REF\_RE.is\_match. | Cache classify\_block result before is\_passthrough\_block check and reuse in the LinkReferenceDefinition branch to eliminate duplicate regex evaluation. |

**Comment by leynos** [comment]
@coderabbitai review

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already.

### ❌ Failed checks (1 error, 2 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Unit Architecture | ❌ Error | classify\_block instantiates LinkReferenceMatcher internally (line 107 of block.rs), violating explicit dependency injection. Line 165 uses non-Oxford "Centralises"; line 183 omits comma before "so". | Inject LinkReferenceMatcher from wrap\_text to classify\_block. Fix "Centralises" → "Centralizes" on line 165 and add comma on line 183 of docs/developers-guide.md. |
| Developer Documentation | ⚠️ Warning | Documentation exists but has two unresolved issues: line 166 uses "Centralises" instead of "Centralizes", and line 183 lacks comma before "so". | Fix spelling: "Centralises" → "Centralizes" (line 166). Fix grammar: add comma before "so" (line 183). Add LinkReferenceMatcher/LinkTitleWindow to API reference table. |
| Testing (Property / Proof) | ⚠️ Warning | Proptest omits critical invariants: single-quoted/parenthesis titles, angle-bracketed URLs, indentation (0-3), escaped content, and state machine transitions not property-tested. | Extend proptest: add generators for all title forms, angle-bracketed URLs, definition indentation, escaped characters, and property tests for state machine transitions. |

**Comment by leynos** [comment]
@coderabbitai review
